### DISCUSSION TITLE
راهنمای فارسی استفاده از برنامه

### DISCUSSION THREAD

**Discussion by ImanMontajabi** [ImanMontajabi/CloudflareDNS#3] [Announcements]
سلام این برنامه  تقریبا یک ابزار درکنار اسکریپت یا برنامه ای هست که [مرتضی باشسیز](https://github.com/MortezaBashsiz/CFScanner) درست کرده برای اسکن آی‌پی های تمیز کلادفلر. هم کاربرای ویندوز و هم لینوکس میتونند از این برنامه استفاده کنند. 
خوب اول از همه شما شروع میکنید با اسکنر مرتضی آی‌پی هارو اسکن کردن مثلا هزارتا آی‌پی تمیز که به ترتیب بهترین به بدترین مرتب شده. نیازی به **Export** کردن نیست اسکنر مرتضی نسخه ویندوزی توی پوشه‌ی **result** به صورت فایل های **json** ذخیره میکنه ما برای ایجاد ساب دامین از اون آی‌پی ها استفاده میکنیم. (برای لینوکس این فایل پسوند **cf.** دارد)
![Screenshot (25)](https://user-images.githubusercontent.com/52942515/227301140-7f3582e5-ebaa-4788-b27f-88a77f84aca4.png)
## امنیت
چک کردن مسائل امنیتی برنامه مثل خوندن سورس کد یا ایجاد فایل اجرایی شخصی، به عهده‌ی خودتونه. اطلاعات ورودی شما تماما در کامپیوتر خودتون کنار فایل برنامه در فایل **user_id.json** ذخیره میشه.
## اجرا
موارد خواسته شده در کادر برنامه رو وارد کنید یه سری فیلم مرتبط هست گذاشتم [اینجا](https://t.me/CloudflareDNS)
----------------------------------------------------------
![Screenshot (60)](https://user-images.githubusercontent.com/52942515/230775628-6de0fcc1-d0fb-4de8-9620-0ee157dbbf0f.png)
![Screenshot (52)](https://user-images.githubusercontent.com/52942515/228668850-fdf4572d-72ba-4c52-9cb2-2a7a3d33ce77.png)
![Screenshot (51)](https://user-images.githubusercontent.com/52942515/228668815-7edc6510-cf5a-4aca-b0e4-ce42e5b9dd3b.png)
---------------------------------------------------------
## مهم
- حداکثر صدتای اول نتیجه‌ی تست [اسکنر](https://github.com/MortezaBashsiz/CFScanner) برای ساب‌دامین استفاده میشه یعنی بین یک تا نهایت صدتای برتر حالا هر تعدادی که میخواد باشه
- و مقدار TTL به صورت خودکار برابر یک قرار خواهد گرفت ولی توی کلادفلر auto نشون میده ولی تست کردم مقدارش برابر یک بود !
- برای این که حین ایجاد نسخه شخصی با warning مواجه نشید از [این آموزش](https://youtu.be/vkKCiEYp6ns) استفاده کنید
- آی‌پی از هر اسکنر دیگه که دارید رو توی یه فایل **txt.** ذخیره کنید طوری که توی هر خط یه آی‌پی باشه ترجیحا، یا خلاصه **آی‌پی ها از هم جدا باشند**. بعد که سیو کردید پسوند فایل رو به **cf.** تغییر بدید حالا از اون‌ها هم میتونید داخل برنامه استفاده کنید.
-----------------------------------------------------------------------
1. **List** : تمام آی‌پی های موجود در ساب دامین داده شده رو لیست میکنه
![aaa](https://user-images.githubusercontent.com/52942515/228953755-8ab13d9b-cff8-4442-9e38-a8d4d181cfe0.jpg)

2. **Create** : 
 **Result file** اول با استفاده از دکمه‌ی 
 فایل نتیجه‌اسکن که گفتم رو از برنامه‌ی مرتضی بگیرید بعد این میاد حداکثر صدتا آی‌پی اول رو میزاره توی ساب دامینی که بهش دادید 
![Screenshot (55)](https://user-images.githubusercontent.com/52942515/228953809-85d4d407-9b00-41a2-9293-4e77280dd609.jpg)
![sfsdfsfsf](https://user-images.githubusercontent.com/52942515/228953820-8ff036fc-50f6-48f8-9e92-a73e4e2a3335.jpg)

3. **Delete** : تمام آی‌پی‌های ساب دامینی که بهش دادید رو پاک میکنه هرچندتا که باشه
![fsfds](https://user-images.githubusercontent.com/52942515/228953861-de3dba17-c194-407e-8bec-5ae066c6c89f.jpg)

4. **Export** : آی‌پی های ساب دامینتون را داخل یک فایل **جی‌سان** خروجی میگیرد و کافی است آن را خل پوشه‌ی **ریزالت** برنامه‌ی اسکنر کپی کنید و مطابق آموزش داخل کانال تلگرام آن‌ها را دوباره اسکن کنید
![222](https://user-images.githubusercontent.com/52942515/229312832-850ca018-02d8-4638-86de-d8854cca158e.png)

5. دراین کادر حداکثر آی‌پی که میخواهید از فایل اسکنتون انتخاب بشه رو تعیین میکنید ولی محدوده‌ی مجاز بین یک تا 100 هستش
![new](https://user-images.githubusercontent.com/52942515/229368441-f6181e2e-86e9-45c6-9223-a321c2e9d7ef.jpg)
-----------------------------------------------

دانلود برنامه
===================
از اینجا میتونید برنامه رو [دانلود](https://github.com/ImanMontajabi/CloudflareDNS/releases/download/v0.3.3/CloudflareDNS.exe) کنید

**Comment by MatinBerimo** [comment]
سلام ایمان جان من همه مراحل رو دو سه بار رفتم ولی وقتی create رو میزنم دو فایل ip و best ip درست میشه اما داخل برنامه اتفاقی نمی افته

![Screenshot 2023-03-24 015407](https://user-images.githubusercontent.com/77873824/227378255-9ddbfe92-d515-4e5a-b875-072f933d7395.png)

**Reply by ImanMontajabi** [reply]
لیست و دیلیت کار میکنن؟؟؟ داخل اون فایل ها که درست میشه آی‌پی هست؟؟؟ توی ip.txt آی‌پی های مرتضی باید باشه و توی ‌best_ip.txt صدتا آی‌پی برتر چک کن نتیجشو بگو

**Reply by MatinBerimo** [reply]
لیست و دیلیت هم کار نمی کنن و این که داخل اون فایل ها آیپی هستش و داخل besr ip صد ایپی هست ولی create رو میزنم کار نمیکنه

**Reply by ImanMontajabi** [reply]
اون اطلاعاتت رو اشتباه وارد کردی به اکانتت وصل نمیشه من مدیریت خطا رو کامل نکردم ولی مشکل همینه

**Comment by Sysop0098** [comment]
کرش میکنه برنامه

**Reply by ImanMontajabi** [reply]
تمام مراحل بالا رو درست بری نمیکنه

**Comment by saeex71** [comment]
حتما باید 100 تا ای پی و بیشتر باشه؟ تعداد کمتر نمیشه؟ و اگه یک فایل .json داشته باشیم که توی هر خط فقط یک ای پی باشه جواب میده؟

**Reply by ImanMontajabi** [reply]
این برنامه بر مبنای اسکنر مرتضی درست شده احساس میکنم زیر صدتا اسکن نتیجه خیلی شانسیه و الزاما آی‌پی های خوبی پیدا نمیشه و مبنا رو گزاشتم روی یه اسکن کامل یا حداقل کافی به نظرم زیر صدتا اسکن لطفی نداره و از همین سایتای آی‌پی تمیز که شانسی آی‌پی خوب هم پیدا میکنه! در عوض میشه استفاده کرد و تعداد کم آی‌پی مثلا سه تا رو دستی هم میشه وارد کرد! هرچند سورس کد رو میتونید دستکاری کنید و هرجا 100 بود به دلخواه تغییر بدید و اگردرخواست زیاد باشه این فیچر تعداد آی‌پی دلخواه رو اضافه میکنم. ضمنا فرمت فایل json جور دیگه ایه پس خیر نمیشه ممنون از شما

**Reply by saeex71** [reply]
حالا اون صدتا هیچی 
من از نسخه لینوکس مرتضی  bash استفاده میکنم بخاطر قابلیت اسکن بیشتر مخصوصا تست اپلود خروجی فایل json نیست و cf هست اینو چیکار کنم؟

**Reply by ImanMontajabi** [reply]
داداش من میخواستم برا لینوکسشو درست کنم من کاربر ویندوزم اون پیشنیازای برنامه رو نتونستم نصب کنم همشو اسکریپت لینوکسی هم اجرا شد ولی هیچ خروجی نداد من بلد نبودم و حوصله لینوکسم نداشتم سر هر کار کسشری سه ساعت سرچ کنم :) اول به من بگو اونو چه جوری راه بندازم لینوکسشم کاری نداره اونم درست میکنم :) همینو برای لینوکس خروجی گرفتم کامل کار میکرد فقط برا همون دلیل که گفتی create کار نمیکرد چون خروجی json نبود

**Reply by ImanMontajabi** [reply]
یه فایل خروجی اسکریپتو همینجا آپلود کن تا برا لینوکسشو اکی کنم

**Reply by saeex71** [reply]
منم مثل شما کاربر ویندوز هستم و از طریق wsl یا شبیه ساز ترمینال لینوکس اوردم بالا و کد مرتضی کلون کردم و تست و خروجی گرفتم
دقیقا مشکل راه اندازی کجا داری؟ ترمینال ؟ پیش نیازها؟ یا ...

**Reply by ImanMontajabi** [reply]
.

**Reply by saeex71** [reply]
اوکی میفرستم

**Comment by HadiDkb** [comment]


**Reply by ImanMontajabi** [reply]
دوتا مسأله هست: اول امنیت و اعتماد کسی که اون برنامه رو روی گوشیش نصب میکنه باید فراهم باشه. دوم مبنای کار این برنامه اسکنر مرتضی باشسیز هست اگه اسکنر برای گوشی وجود داشته باشه حتما برای اون هم من انجام میدم

**Reply by HadiDkb** [reply]
سپاس فراوان و ممنون که وقت گذاشتین و جواب دادین🙏

**Comment by PikachuJP** [comment]
Traceback (most recent call last):
  File "CloudflareDNS.py", line 1, in <module>
ImportError: DLL load failed while importing QtGui: The specified procedure could not be found.

**Reply by PikachuJP** [reply]
On app start Shows this error

**Reply by ImanMontajabi** [reply]
Try to make your own executable version from the source code, just make sure to use this command : pip install -r requirements.txt --use-pep517
Also, the third video of the Telegram channel is related to the same issue
Sometimes Windows  DLLs need to be repaired, you can use DLL fixer, google it :)

**Reply by PikachuJP** [reply]
Thanks

**Comment by PikachuJP** [comment]
Update
You don't need to use this program for getting fresh and clean IPs
You just have to make a CloudFlare worker and attach your domain to the worker in trigger tab add custom domain
And everything good to go automatic simple and fast

**Reply by ImanMontajabi** [reply]
Excellent, just what do you enter the address-field in vmess or vless config ??!!
We tried it, we need clean Cloudflare IPs

**Reply by PikachuJP** [reply]
Just make simple worker (That default hello world script is OK don't need to change the source) and add custom domain to it after that just make your subscription link like this also I guess you don't need to use custom domain too I didn't test with workers.dev domain but here's the example

https://xxxx.xxxx.workers.dev/sub/(CloudFlare worker domain or custom CloudFlare worker domain)

**Reply by PikachuJP** [reply]
I'll make complete guide soon I hope you understand from this example

Update : I guess you can use the sub maker worker url as clean ip too (not checked yet)

**Reply by PikachuJP** [reply]
Also you can use this source if you want clean ip I've edited the source for better quality also ip range updated but you may get some lags on start full range scan but this new method works good too

https://cfip.int.eu.org

**Reply by ImanMontajabi** [reply]
Thank you very much, I will definitely test it, you did a great job, but we had already tested this :
- https://github.com/vfarid/cf-ip-scanner
- https://github.com/vfarid/cf-ip-scanner-py

**Reply by PikachuJP** [reply]
Yup I've checked that is awesome great job

**Comment by sina1100** [comment]
ایمان جان دمت گرم ، میشه لینک برنامه آقا مرتضی را  هم بگذاری 
ممنون از زحماتتون همگیتون 🙌

**Reply by ImanMontajabi** [reply]
سلام. لطف دارید
https://github.com/MortezaBashsiz/CFScanner/releases/tag/v1.1.8489.35707

**Comment by PikachuJP** [comment]
![Screenshot](https://user-images.githubusercontent.com/128409341/229488703-3b985b4f-431b-48d9-9f22-a34155bdb86d.jpg)

!!!

**Comment by ImanMontajabi** [comment]
First you should install requirements:
pip install -r requirements.txt --use-pep517

**Reply by PikachuJP** [reply]
I did that before but it gives same error

**Reply by ImanMontajabi** [reply]
You clearly did not install the packagese, probably you did not choose the right interpreter. Try this too:
pip install pyqt6
pip install requests
pip install auto-py-to-exe

**Reply by PikachuJP** [reply]
![image](https://user-images.githubusercontent.com/128409341/229583350-0d2fb39a-8822-4fce-a8c4-99c87162cbc4.png)

Look

**Reply by ImanMontajabi** [reply]
Everything is correct. Now you have to run the program. The problem is with your vs code, which cannot recognize your python packages. My code has no problem.
ctrl+shift+p > select interpreter

**Reply by PikachuJP** [reply]
I don't have access to my system rn, it's probably from github codespace

**Comment by AliShFar** [comment]
سلام و درود
وقتتون بخیر
من همه کارها رو انجام میدم. به کلود هم وصل میشم. چون لیست که میگیرم رکوردی که از قبل ساختم رو میاره
ولی نه دیلیت میکنه و نه رکورد جدیدی اضافه میشه
فقط رکوردهای قبلی رو نشون میده
همین
مشکل کجاست؟

**Reply by ImanMontajabi** [reply]
اگه list رو که میزنید رکورد های قبلی رو نشون میده یعنی اطلاعات شما درست بوده بلافاصله delete رو که بزنید باید رکوردهارو پاک کنه مطابق کلیپ کانال تلگرام و قبل این که create رو بزنید با result file فایل اسکن رو بدید به برنامه و مطمئن بشید خالی نیست. روشش همینه من توی تست هایی که میگیرم مشکلی نیست واکثر دوستان هم مشکلی نداشتند. مثل فیلمی که توی کانال هست و گرنه رفعش میکردم. مطمئن بشید اینترنتتون وصله اگر در نهایت باز هم نشد مثل کلیپ سوم کانال یا با پایتون اجرا کنید برنامه رو یا برای خودتون نسخه اجرایی بسازید .

**Reply by AliShFar** [reply]
همه مراحل رو همینجوری که توضیح داده بودید انجام دادم ولی متاسفانه نتیجه نگرفتم
ضمنا از نسخه پایتون هم استفاده کردم
نسخه ویندوز که کرش میکنه و بسته میشه

**Reply by ImanMontajabi** [reply]
من دوباره تست کردم مشکلی نداشت فیلم هم گرفتم همین آخرین نسخه رو از گیت هاب دانلود کردم انجام دادم نه کرش کرد نه مشکلی داشت بسته‌های c++ و .net رو از سایت soft98 دانلود و نصب کنید و ویندوزتون رو آپدیت کنید روی ماشین مجازی یه ویندوز نصب کردم و تست کردم باز هم مشکلی نداشت

**Comment by Hooti66** [comment]
خدا وکیلی دمت گرم عالیه کارم خیلی راحت شد با یه فایل اکسل کنارش دارم ازش استفاده میکنم خیلی خیلی راحت شد کارم آفرین به تو

**Reply by ImanMontajabi** [reply]
قربانت :))))
